### ISSUE TITLE
Unnecessary GuidGenerator calls for custom delegates in authored types

### ISSUE THREAD

**Issue by Sergio0694** [microsoft/CsWinRT#1499]
**Describe the bug**

We're emitting some `GuidGenerator` calls when using custom delegates in authored components:

![image](https://github.com/microsoft/CsWinRT/assets/10199417/d8b61325-1e22-45c5-87c2-86c3d5261baf)

Pretty sure we could just literally hardcoded that same GUID we added on the ABI type for the delegate instead?

**To Reproduce**

```csharp
public sealed unsafe partial class HelloWorldEffect
{
    public event MyEventHandler Foo;
}

public delegate void MyEventHandler(object sender, string args);
```

**Expected behavior**

We should probably emit that IID in a static readonly field, and then reference it wherever needed.

**Version Info**

2.0.5-prerelease.240203.7

**Comment by manodasanW** [comment]
This is probably getting optimized by IIDOptimizer today, but I agree we can move this to a readonly static field.

**Comment by Sergio0694** [comment]
If that's the case I think we should probably also fix #1478? Because otherwise IIDOptimizer will break in libraries, and if you just opt-out, then that path will not be optimized by it. I mean I agree here we can just update the code writer, but this just makes me think that in general, the IIDOptimizer _should_ also run on authored libraries then? 🤔
